### PR TITLE
[IOT] Test updates for beta branch

### PR DESCRIPTION
**Pull request by vilit1**
**Description**

Testing updates, specifically rerecording tests and ensuring that commands would work.
https://github.com/Azure/azure-cli/issues/22174

Additional Changes:
 - _test_utils.py was changed for all versions to use cryptography over pyopenssl.
 - Updated the certificate create code to serialize the correct body for non-latest profiles

Tests not rerecorded:
 - DPS and Central are not listed in AZURE_API_PROFILES 2020-09-01-hybrid profile and thus have not been re-recorded
 - IoT Hub identity test was not originally recorded in the 2020-09-01-hybrid profile and is thus not re-recorded.

**Testing Guide**
Used `azdev test <command> --profile <profile> --live` to rerecord tests

**History Notes**
[IOT] Test updates for beta branch

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
IoT

File: src/azure-cli/azure/cli/command_modules/iot/tests/hybrid_2020_09_01/_test_utils.py [hunk 1]
```diff
@@ -3,78 +3,90 @@
...
+        )
+        .sign(key, hashes.SHA256())
+    )
+
+    key_dump = key.private_bytes(
+        encoding=serialization.Encoding.PEM,
+        format=serialization.PrivateFormat.TraditionalOpenSSL,
+        encryption_algorithm=serialization.NoEncryption(),
+    ).decode("utf-8")
+    cert_dump = cert.public_bytes(serialization.Encoding.PEM).decode("utf-8")
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/iot/tests/hybrid_2020_09_01/_test_utils.py:44]: Maybe this procedure can be extracted to a common function. 😉

**Review by jiasli — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/iot/tests/hybrid_2020_09_01/_test_utils.py (hunk 1 above)
Comment (vilit1) [reply src/azure-cli/azure/cli/command_modules/iot/tests/hybrid_2020_09_01/_test_utils.py:44]: added to backlog

File: src/azure-cli/azure/cli/command_modules/iot/tests/latest/test_iot_commands.py [hunk 2]
```diff
@@ -428,7 +428,7 @@ def test_iot_hub(self, resource_group, resource_group_location, storage_account)
                          self.check('properties.enableDataResidency', True)])
         self.cmd('az iot hub delete -n {}'.format(dr_hub_name))
 
-    @unittest.skip('Will be recorded in https://github.com/Azure/azure-cli/pull/22262')
+    # @unittest.skip('Will be recorded in https://github.com/Azure/azure-cli/pull/22262')
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/iot/tests/latest/test_iot_commands.py:431]: A quick question, can this test be live-run by anyone on any subscription? If not, we should mark it as live-only.

**Comment by zhoxing-ms** [comment]
Your PR should not modify so many modules. Please rebase pull and rebase code from the remote dev branch first
Screenshot 2022-05-18 202519

File: src/azure-cli/azure/cli/command_modules/iot/tests/latest/test_iot_commands.py (hunk 2 above)
Comment (vilit1) [reply src/azure-cli/azure/cli/command_modules/iot/tests/latest/test_iot_commands.py:431]: This should be runnable by anyone (as long as they have permissions to assign identities to the hub)

**Review by jiasli — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**